Thumbprint/Targets/Thumbprint/Components/Pill.swift

```diff
@@ -231,8 +235,8 @@ open class Pill: UIView, UIContentSizeCategoryAdjusting {
         stackView.addArrangedSubview(label)
 
         iconImageView.snp.makeConstraints { make in
-            make.height.equalTo(Icon.Size.tiny.dimension)
-            make.width.equalTo(Icon.Size.tiny.dimension).priority(.high)
+            make.height.equalTo(config.iconSize.dimension)
```

Comment:
Do you know if `iconSize` can be updated after initialization? I'm not 100% sure if the constraint would be updated to reflect that change

Reply:
I am not sure as my use case didn't have a requirement of changing icon sizes I didn't try it.

Reply:
Gotcha. Since there is a config this might not happen, but the use case that comes to mind is view setup

```
private let pill = Pill()
...
func setupViews() {
    pill.iconSize = ...
}
```

Reply:
Thats what I was thinking. Only way to know for sure would be to do something like this
```

class SomeView {
  
@objc func someButtonPressed() {

UIView.animate () {
    pill.config = Pill.Config(height:....,iconSize: Icon.Size.Tiny) // animate change of icon size to tiny size.
    self.layoutIfNeeded()
}
}

}
```

Reply:
Ah thought a little more about this and I see what you mean - because the icon size constraint is invoked in the initializer, it won't get called if you set a config afterwards For example:

```
let pill = Pill(adjustsFontSize...)
pill.config = Pill.Config(height:...,iconSize: Icon.Size.medium) /// this won't change the icon to medium 
```

Looking into this.